src/libraries/Common/tests/TestUtilities/System/PlatformDetection.Unix.cs

```diff
@@ -204,11 +153,24 @@ private static Version ToVersion(string versionString)
             }
         }
 
-        private static DistroInfo GetDistroInfo() => new DistroInfo()
+        private static DistroInfo GetDistroInfo()
         {
-            Id = Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment.OperatingSystem,
-            VersionId = ToVersion(Microsoft.DotNet.PlatformAbstractions.RuntimeEnvironment.OperatingSystemVersion)
-        };
+            DistroInfo result = new DistroInfo();
+
+            foreach (string line in File.ReadAllLines("/etc/os-release"))
+            {
+                if (line.StartsWith("ID=", StringComparison.Ordinal))
+                {
+                    result.Id = line.Substring(3).Trim('"', '\'');
+                }
+                else if (line.StartsWith("VERSION_ID=", StringComparison.Ordinal))
+                {
+                    result.VersionId = ToVersion(line.Substring(11).Trim('"', '\''));
+                }
+            }
+
+            return result;
```

Comment:
do we need to ensure we don't have a null Id or version inside the result?

Reply:
Done.